crates/ruff_linter/src/rules/flake8_unused_arguments/rules/unused_arguments.rs

```diff
@@ -311,6 +311,63 @@ fn call<'a>(
     }));
 }
 
+/// Returns `true` if a function appears to be a base class stub. In other
+/// words, if it matches the following syntax:
+///
+/// ```text
+/// variable = <string | f-string>
+/// raise NotImplementedError(variable)
+/// ```
+///
+/// See also [`is_stub`]. We're a bit more generous in what is considered a
+/// stub in this rule to avoid clashing with [`EM101`].
+///
+/// [`is_stub`]: function_type::is_stub
+/// [`EM101`]: https://docs.astral.sh/ruff/rules/raw-string-in-exception/
+fn is_not_implemented_stub_with_variable(function_def: &StmtFunctionDef) -> bool {
+    // Ignore doc-strings.
+    let statements = match function_def.body.as_slice() {
+        [Stmt::Expr(StmtExpr { value, .. }), rest @ ..]
+            if matches!(**value, ast::Expr::StringLiteral(_)) =>
+        {
+            rest
+        }
+        _ => &function_def.body,
+    };
+
+    let [Stmt::Assign(ast::StmtAssign { value, .. }), Stmt::Raise(StmtRaise {
+        exc: Some(exception),
+        ..
+    })] = statements
+    else {
+        return false;
+    };
+
+    if !matches!(
+        value.as_ref(),
+        ast::Expr::StringLiteral(_) | ast::Expr::FString(_)
+    ) {
+        return false;
+    }
+
+    let ast::Expr::Call(ast::ExprCall {
+        func, arguments, ..
+    }) = &**exception
+    else {
+        return false;
+    };
+
+    if !matches!(&**func, ast::Expr::Name(name) if name.id == "NotImplementedError") {
+        return false;
+    }
+
+    if arguments.len() != 1 {
+        return false;
+    }
```

Comment:
I think we'd also need to make sure that the variable that is being referenced in the call to `NotImplementedError` is the same as the one in the assignment. I think there must be some method in the `SemanticModel` to get the binding for the `ExprName` node (possibly `resolve_name`?).

Or, actually we can get away with just comparing the string i.e., whether the name of the variable that is being assigned is the same as the one used in the `NotImplementedError` call because we make sure that there are only two statements in the function body above.